packages/store/ts/config/storeConfig.ts

```diff
@@ -88,7 +88,18 @@ export function resolveUserTypes<
 >(schema: TSchema, userTypes: TUserTypes): ResolvedSchema<TSchema, TUserTypes> {
   const resolvedSchema: Record<string, SchemaAbiType> = {};
   for (const [key, value] of Object.entries(schema)) {
-    resolvedSchema[key] = (userTypes[value]?.internalType as SchemaAbiType) ?? value;
+    const staticArray = parseStaticArray(value);
+
+    if (userTypes[value] !== undefined) {
+      // user-defined types
+      resolvedSchema[key] = userTypes[value].internalType as SchemaAbiType;
+    } else if (staticArray) {
+      // static arrays
+      resolvedSchema[key] = `${staticArray.elementType as StaticAbiType}[]`;
+    } else {
+      // abi types
+      resolvedSchema[key] = value as SchemaAbiType;
+    }
```

Comment:
```suggestion
    const staticArray = parseStaticArray(value);

    if (isSchemaAbiType(value)) {
      resolvedSchema[key] = value;
    else if (userTypes[value] !== undefined) {
      resolvedSchema[key] = userTypes[value].internalType as SchemaAbiType;
    } else {
      const staticArray = parseStaticArray(value);
      if (!staticArray) throw new Error(`Unexpected type: ${value}`);
      resolvedSchema[key] = `${staticArray.elementType as StaticAbiType}[]`;
    }
```

Reply:
oh actually, I think this file deals with old schema ABI type abstractions, so that may not work

Reply:
added `isSchemaAbiType` and imported and used the suggestion